flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/ResourceManagerFactory.java

```diff
@@ -59,4 +63,16 @@ default String generateEndpointIdWithUUID() {
 	default String getEndpointId() {
 		return ResourceManager.RESOURCE_MANAGER_NAME;
 	}
+
+	default Configuration getResourceManagerConfiguration(Configuration flinkConfig) {
```

Comment:
Why do we make this method part of the `ResourceManagerFactory`? It seems to me that we are mixing concerns here (creating a `ResourceManager` vs. properly setting up the ResourceManager's `configuration`).